test/Board.cpp

```diff
@@ -77,3 +92,154 @@ TEST_CASE("Mixed INO/C++ sources", "[Board]") {
         std::cerr << tc.build_log().second;
     REQUIRE_FALSE(ec);
 }
+
+/**
+ * Test that attaching sketches to the board and resetting of the board only works
+ * when the board is not running and not suspended.
+ */
+TEST_CASE("Board - Check conditions for attach_sketch and reset", "[Board]") {
+    smce::Toolchain tc{SMCE_PATH};
+    REQUIRE(!tc.check_suitable_environment());
+
+    smce::Sketch sk{SKETCHES_PATH "noop", {.fqbn = "arduino:avr:nano"}};
+    tc.compile(sk);
+    REQUIRE(sk.is_compiled());
+
+    smce::Sketch sk2{SKETCHES_PATH "noop", {.fqbn = "arduino:avr:nano"}};
+    tc.compile(sk2);
+    REQUIRE(sk2.is_compiled());
+
+    // Board is clean, so not running yet => Attaching sketches and resetting possible
+    smce::Board br{};
+    REQUIRE(br.status() == smce::Board::Status::clean);
+    REQUIRE(br.attach_sketch(sk));
+    REQUIRE(br.reset());
+
+    // Board is configured, so not running yet => Attaching sketches and resetting possible
+    REQUIRE(br.configure({}));
+    REQUIRE(br.status() == smce::Board::Status::configured);
+    REQUIRE(br.attach_sketch(sk));
+    REQUIRE(br.reset());
+
+    // Board is running => Attaching sketches and resetting NOT possible
+    REQUIRE(br.configure({}));
+    REQUIRE(br.attach_sketch(sk));
+    REQUIRE(br.start());
+    REQUIRE(br.status() == smce::Board::Status::running);
+    REQUIRE_FALSE(br.attach_sketch(sk2));
+    REQUIRE_FALSE(br.reset());
+
+    // Board is suspended => Attaching sketches and resetting NOT possible
+    REQUIRE(br.suspend());
+    REQUIRE(br.status() == smce::Board::Status::suspended);
+    REQUIRE_FALSE(br.attach_sketch(sk2));
+    REQUIRE_FALSE(br.reset());
+
+    // Board is stopped => Attaching sketches and resetting possible
+    REQUIRE(br.stop());
+    REQUIRE(br.status() == smce::Board::Status::stopped);
+    REQUIRE(br.attach_sketch(sk2));
+    REQUIRE(br.reset());
+}
+
+/**
+ * Test that configuring the board only works
+ * when the board is clean or already configured.
+ */
```

Comment:
Does "Clean" mean that the board is not yet configured? I.e it is possible to configure the board if the board is not yet configured or has already been configured?

Reply:
It means that you can only configure the board, if:

- The board has the status configured, so the first configuration was done just before.
- The board is clean, so in the initial state. For example, a board that has already run, cannot be configured again. Instead, it has to be cleaned/reset first.

Reply:
I will improve the comment in the test case.

Reply:
Looks good now